File: src/main/java/no/nav/pto/veilarbfilter/domene/FilterModel.java
package no.nav.pto.veilarbfilter.domene;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import no.nav.pto.veilarbfilter.domene.deserializer.DateDeserializer;
import no.nav.pto.veilarbfilter.domene.deserializer.DateSerializer;

import java.time.LocalDateTime;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
@AllArgsConstructor
@NoArgsConstructor
public class FilterModel {
    private Integer filterId = null;
    private String filterNavn = null;
    private PortefoljeFilter filterValg = null;
    @JsonDeserialize(using = DateDeserializer.class)
    @JsonSerialize(using = DateSerializer.class)
    private LocalDateTime opprettetDato = null;
    private Integer filterCleanup = 0;
}

File: src/main/kotlin/no/nav/pto/veilarbfilter/model/FilterModel.kt
package no.nav.pto.veilarbfilter.model

import org.jetbrains.exposed.sql.Table
import org.jetbrains.exposed.sql.datetime
import java.time.LocalDateTime


object EnhetFilter : Table() {
    val filterId = integer("filter_id").primaryKey().autoIncrement()
    val enhet = varchar("enhet_id", 32)
    val filterNavn = varchar("filter_navn", 255)
    val valgteFilter = jsonb("valgte_filter", EnhetPortefoljeFilter::class.java)
    val opprettetDato = datetime("opprettet")
}

data class FilterModel (
    val filterId: Int,
    val filterNavn : String,
    val filterValg: EnhetPortefoljeFilter
)

data class EnhetFilterModel (
    val filterId: Int,
    val enhetId: String,
    val filterNavn: String,
    val filterValg: EnhetPortefoljeFilter,
    val opprettetDato: LocalDateTime
)

data class NyttFilterModel (val filterNavn : String, val filterValg: EnhetPortefoljeFilter)
